OpenTrons/opentrons-api
# feat(protocol-designer): add thermocycler form profile cycle functionality

# Overview

Adds the ability to program thermocycler profile cycle steps and wires up to thermocycler step form.

TODO: add tests

Closes AUTH-810

## Test Plan and Hands on Testing

- open PD redesign and create thermocycler step
- verify that radio button shows for changing state or programming a profile
- select program profile and continue
- Sselect button under "profile steps" to open modal
- add cycles and steps, and ensure you can add, edit, and remove steps/cycles properly. Modal and step/cycle save buttons should be disabled if form fields are in a bad state
- save the modal and step form and export protocol
- verify that Thermocycler steps appear in commands as expected
- reupload the downloaded protocol to PD
- select the programmed thermocycler step and verify that you land on step 2 of the form, with all selected fields and profile pre-populated

## Changelog

- create `ThermocyclerCycle` component and wire up in modal
- refactor types
- add logic for entering thermocycler step form on step 2 if values are populated

## Review requests

see test plan

## Risk assessment

low
